packages/context/src/__tests__/acceptance/binding-config.acceptance.ts

```diff
@@ -0,0 +1,192 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {config, configBindingKeyFor, Context, ContextView} from '../..';
+
+describe('Context bindings - injecting configuration for bound artifacts', () => {
+  let ctx: Context;
+
+  beforeEach(givenContext);
+
+  it('binds configuration independent of binding', async () => {
+    // Bind configuration
+    ctx.configure('servers.rest.server1').to({port: 3000});
```

Comment:
The way how this test is written, it's not clear at all that RestServer is using `@config` with no binding key set, therefore it will pick up configuration associated with the current RestServer binding being resolved.

Let's add a new test to verify this specific scenario.

```ts
class RestServer {
  constructor(
    // Notice there is no binding key specified for the @config() decorator
    // @config will pick configuration binding associated with the binding key 
    // of the server being instantiated (resolved)
    @config() public configObj: RestServerConfig
  ) {}
}

ctx.bind('server1').toClass(RestServer);
ctx.bind('server2').toClass(RestServer);

ctx.configure('server1').to({port: 3000});
ctx.configure('server2').to({port: 8000});

// server1 picks up config for server1
const server1 = ctx.get<RestServer>('server1');
expect(server1.configObj).to.eql({port: 3000});

// server2 picks up config for server2
const server2 = ctx.get<RestServer>('server2');
expect(server2.configObj).to.eql({port: 8000});
```

Reply:
Maybe the test I am asking for is already implemented as `it('injects a config property based on the parent binding'`?

Reply:
Yes.